Documentation/ModuleAliasing.md

```diff
@@ -0,0 +1,156 @@
+# Module Aliasing by SwiftPM
+
+## Overview
+
+The number of package dependencies often grows, with that, a name collision can occur among modules from different packages. Module names such as `Logging` or `Utils` are common examples. In order to resolve the collision, SwiftPM (in 5.7+) introduces a new parameter `moduleAliases`, which allows a user to define new unique names for the conflicting modules without requiring any source code changes.  
+
+## How to Use
+
+Let's consider the following scenarios to go over how module aliasing can be used. 
+
+### Example 1
+
+`App` imports a module called `Utils` from a package `swift-draw`. It wants to add another package dependency `swift-game` and imports a module `Utils` vended from the package.
+
+```
+ App
+   |— Module Utils (from package ‘swift-draw’)
+   |— Module Utils (from package ‘swift-game’)
+```
+
+Package manifest `swift-game`
+```
+{
+    name: "swift-game",
+    products: [
+        .library(name: "UtilsProduct", targets: ["Utils"]),
```

Comment:
maybe this example can be further simplified if the product would also be called "Utils" (instead of "UtilsProduct")?

Reply:
also, if we are showing the manifest for `swift-game`, we should probably do the same for ‘swift-draw’ to make the example complete

Reply:
This is intentional; explanation is at L54.

Reply:
Modifying duplicate product names for dependency resolution might be nontrivial based on discussions around this; currently tracked at rdar://89836609 as a candidate for v2 (cc @neonichu)

Reply:
okay, we can discuss this more off thread